pkg/api/v1beta3/types.go

```diff
@@ -206,7 +206,10 @@ type HostPathVolumeSource struct {
 	Path string `json:"path" description:"path of the directory on the host"`
 }
 
-type EmptyDirVolumeSource struct{}
+type EmptyDirVolumeSource struct {
+	// Optional: make this empty directory be backed by memory.
+	Tmpfs bool `json:"tmpfs" description:"use tmpfs instead of disk"`
```

Comment:
I would be okay going with type for now which we could later extend for
flash. @erictune put that idea forth before and an earlier version of the
kubelet tmpfs work did something like that.
On Mon, Mar 9, 2015 at 7:59 PM Brian Grant <notifications@github.com> wrote:

> In pkg/api/v1beta3/types.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5166#discussion_r26088686>
> :
>
> > @@ -206,7 +206,10 @@ type HostPathVolumeSource struct {
> >  	Path string `json:"path" description:"path of the directory on the host"`
> >  }
> >
> > -type EmptyDirVolumeSource struct{}
> > +type EmptyDirVolumeSource struct {
> > +	// Optional: make this empty directory be backed by memory.
> > +	Tmpfs bool `json:"tmpfs" description:"use tmpfs instead of disk"`
>
> Why not type -- tmpfs, disk -- which we could extend later to ssd?
>
> We've also discussed using requirements (latency, etc.) rather than type,
> but I don't know that's really viable.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5166/files#r26088686>
> .
>